### PR TITLE
Post create

### PR DESCRIPTION
**Pull request by linse**
Adding missing methods post_is_create and create_path.

### CODE REVIEW

**Comment by seliopou** [comment]
I'm finally getting around to looking at this.

**Comment by seliopou** [comment]
So it looks like this doesn't fully implement POST support, as it's missing a few decision points and doesn't actually make use of `allow_missing_post` which is still commented out. I don't know if it's safe to just modify `n11` without considering all the other decision points.

**Comment by hannesm** [comment]
Hi Spiros,

thanks for looking into this PR.

> So it looks like this doesn't fully implement POST support, as it's missing a few decision points and doesn't actually make use of allow_missing_post which is still commented out.

We just implemented the parts we needed so far, the code for "allow_missing_post" can be added when needed. We are using this PR in an internal project, and it works really well. There, we have tests for create_path and post_is_create.

> I don't know if it's safe to just modify n11 without considering all the other decision points

We followed the Erlang implementation, n11 is inspired by https://github.com/webmachine/webmachine/blob/1.11.1/src/webmachine_decision_core.erl#L459
A test case for n11 is part of this PR.

Which other decision points do you think need to be considered? Can you give us a list so we don't miss anything?

**Comment by seliopou** [comment]
I'd like for any merged PR that adds POST functionality to cover any and all decision points that involve the following callbacks:

* `post_is_create` 
* `allow_missing_post` 
* `create_path`

@tmcgilchrist started implementing some of these in https://github.com/inhabitedtype/ocaml-webmachine/commit/6ff58d817e065eff5eefd3980880c1a26c41a17c. I don't know to what extent that patch covers all the decision points but it's a start.

**Comment by linse** [comment]
Our changes were orthogonal to @tmcgilchrist 's. We combined them, ported the according tests from Erlang and fixed a bug. We think it's good to go now.

**Comment by seliopou** [comment]
I can't seem to push to the pr branch. Could you enable that?

**Comment by linse** [comment]
> I can't seem to push to the pr branch. Could you enable that?

I think you already pushed a commit, and I can't find an option to enable pushing specifically. You could also open a new PR to this branch if you are still stuck. 

Thanks for reviewing! Do you need anything else from us or is this good to go?

File: lib/webmachine.ml [hunk 1]
```diff
@@ -788,17 +789,16 @@ module Make(IO:IO)(Clock:CLOCK) = struct
       self#d "v3l5";
       self#run_op resource#moved_temporarily
       >>~ function
-        | None     -> self#halt 410
+        | None     -> self#v3m5
```
[the remark this answers is not in the document]
Comment (linse) [reply lib/webmachine.ml:792]: Yes. Now it corresponds to the webmachine diagram.

**Comment by seliopou** [comment]
That commit was added via GitHub's web interface. When I try to push I get the following error:

```
$ git push roburio post-create 
ERROR: Permission to roburio/ocaml-webmachine.git denied to seliopou.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by hannesm** [comment]
@seliopou that is strange, we're missing the checkbox to "allow edits from maintainers"; I can see two options: use a branch on your fork (and open a new PR), or PR to our repository.

**Comment by hannesm** [comment]
@seliopou to unblock you on this PR, I invited you with write permissions to roburio/ocaml-webmachine. when you accept that invite, you should be able to push to the branch and complete this PR.

**Comment by hannesm** [comment]
dear @seliopou, a friendly ping on this issue. it would be nice to get this into webmachine-HEAD, and get a release to opam-repository. if you need anything from us, please let us know. thanks.

**Comment by seliopou** [comment]
Sorry forgot about this. I'm gonna do a pass of all my open source libs this weekend, and will look at it then.

**Pull request merged**